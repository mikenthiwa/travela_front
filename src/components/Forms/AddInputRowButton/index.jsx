import React from 'react';
import PropTypes from 'prop-types';
import addButtonIcon from '../../../images/add.svg';
import add from '../../../images/icons/new-request-icons/add.svg';

const AddInputRowButton = ({ text, addRowHandler, wrapperClassName, buttonClassName }) => (
  <div className={wrapperClassName} role="button" tabIndex="0">
    <button type="button" className={buttonClassName} onClick={addRowHandler}>
      <img
        src={(text === 'Add another trip' ? add : addButtonIcon)}
        alt=""
        className="addsvg" />
      {text}
    </button>
  </div>
);

AddInputRowButton.propTypes = {
  text: PropTypes.string.isRequired,
  addRowHandler: PropTypes.func.isRequired,
  wrapperClassName: PropTypes.string.isRequired,
  buttonClassName: PropTypes.string.isRequired,
};

export default AddInputRowButton;
