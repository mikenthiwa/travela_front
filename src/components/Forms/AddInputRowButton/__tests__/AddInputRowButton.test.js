import React from 'react';
import AddInputRowButton from '..';

describe('<AddInputRowButton />', () => {
  const props = {
    text: 'Add a new input row',
    addRowHandler: () => {},
    wrapperClassName: 'input-row-class',
    buttonClassName: 'button-class'
  };
  const wrapper = shallow(<AddInputRowButton {...props} />);

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
