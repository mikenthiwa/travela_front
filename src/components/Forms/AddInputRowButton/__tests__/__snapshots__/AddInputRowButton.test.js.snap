// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddInputRowButton /> renders 1`] = `
<div
  className="input-row-class"
  role="button"
  tabIndex="0"
>
  <button
    className="button-class"
    onClick={[Function]}
    type="button"
  >
    <img
      alt=""
      className="addsvg"
      src="add.svg"
    />
    Add a new input row
  </button>
</div>
`;
